proposals/lip-0052.md

```diff
@@ -882,90 +1287,219 @@ getNextAvailableCollection():
 This function creates an NFT.
 
 ```python
-create(address, collection, attributes):
-    if (length(attributes) > MAX_BYTE_LENGTH_ATTRIBUTES bytes
-        or collectionStore(collection) does not exist):
-        create fails
+def create(
+    address: Address,
+    collection: uint32,
+    attributes: bytes
+    ) -> None:
+
+    if length(attributes) > MAX_BYTE_LENGTH_ATTRIBUTES bytes:
+        emitFailedCreateEvent(address, collection, attributes, RESULT_ATTRIBUTES_TOO_LONG)
+        raise Exception("Atrributes exceed the maximum byte length.")
+    if collectionStore(collection) does not exist:
+        emitFailedCreateEvent(address, collection, attributes, RESULT_COLLECTION_DOES_NOT_EXIST)
+        raise Exception("Collection substore entry does not exist.")
 
     index = collectionStore(collection).nextAvailableIndex
-    nftID = {"chainID": CHAIN_ID_ALIAS_NATIVE,
-             "collection": collection,
-             "index": index}
-    createNFTEntry(nftID, address, attributes)   
+    nftID = CHAIN_ID_ALIAS_NATIVE + uint32be(collection) + uint64be(index)
+    createNFTEntry(nftID, address, attributes)
     createUserEntry(address, nftID)
     collectionStore(collection).nextAvailableIndex += 1
+
+    def emitFailedCreateEvent(
+        address: Address,
+        collection: uint32,
+        attributes: bytes,
+        result: uint32
+    ) -> None:
+
+    emitPersistentEvent(
```

Comment:
Here and in other places where you define `emitFailedEvent` functions, the indentation is wrong. The `emitPersistentEvent` function should be moved right by 1 tab.

Reply:
Very good, thanks!